llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -2043,6 +2044,7 @@ SDValue DAGCombiner::visit(SDNode *N) {
   case ISD::FP_TO_FP16:         return visitFP_TO_FP16(N);
   case ISD::FP16_TO_FP:         return visitFP16_TO_FP(N);
   case ISD::FP_TO_BF16:         return visitFP_TO_BF16(N);
+  case ISD::BF16_TO_FP:         return visitFP16_TO_FP(N);
```

Comment:
Should combine the cases with the other call to visitFP16_TO_FP. I did have to check the body to make sure this was OK, so should also rename the function to reflect it handles both cases